services/google_drive_service.py

```diff
@@ -2,39 +2,58 @@
 import json
 import os
 from typing import Optional
+from urllib.parse import urlparse, parse_qs
 from googleapiclient.discovery import build
 from googleapiclient.http import MediaIoBaseDownload
 from googleapiclient.errors import HttpError
 from google.oauth2.service_account import Credentials
+from services.ssm_service import SSMService
 
-SERVICE_ACCOUNT_FILE = os.environ['GOOGLE_SERVICE_ACCOUNT_FILE']
 
 from logger import create_log
 
 logger = create_log(__name__)
 
-service_account_info = json.loads(SERVICE_ACCOUNT_FILE)
-scopes = ['https://www.googleapis.com/auth/drive']
-credentials = Credentials.from_service_account_info(service_account_info, scopes=scopes)
-
-drive_service = build('drive', 'v3', credentials=credentials)
-
-def get_drive_file(file_id: str) -> Optional[BytesIO]:
-    request = drive_service.files().get_media(fileId=file_id)
-    file = BytesIO()
-
-    try:
-        downloader = MediaIoBaseDownload(file, request)
-
-        done = False
-        while not done:
-            status, done = downloader.next_chunk()
-
-    except HttpError as error:
-        logger.warning(f"HTTP error occurred when downloading Drive file {file_id}: {error}")
-        return None
-    except Exception as err:
-        logger.exception(f"Error occurred when downloading Drive file {file_id}")
-        return None
-
-    return file
+class GoogleDriveService:
+    def __init__(self):
+        ssm_service = SSMService()
+        if os.environ['ENVIRONMENT'] == 'production':
+            SERVICE_ACCOUNT_FILE = ssm_service.get_parameter('arn:aws:ssm:us-east-1:946183545209:parameter/drb/production/google-drive-service-key')
+        else:
+            SERVICE_ACCOUNT_FILE = ssm_service.get_parameter('arn:aws:ssm:us-east-1:946183545209:parameter/drb/qa/google-drive-service-key')
```

Comment:
Would it be better to encapsulate the ARNs in e.g. a dictionary within the SSM service itself?

Reply:
could you do`ssm_service.get_parameter(f'arn.../{os.environ.get('ENVIRONMENT', 'qa')}...` to make this easier?